src/ucp/tag/tag_send.c

```diff
@@ -189,55 +189,95 @@ ucp_tag_send_inline(ucp_ep_h ep, const void *buffer, size_t count,
     return status;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_ptr_t
+ucp_tag_send_common(ucp_ep_h ep, const void *buffer, size_t count,
```

Comment:
i would not introduce ucp_tag_send_common: just implement tag_send_nbx / tag_send_sync_nbx directly
and old functions will call tag_send_nbx / tag_send_sync_nbx

Reply:
ok, re-factored

Reply:
also moved UCP_REQUEST_FLAG_SYNC into inside attr_mask